appstore/developer_portal_api.py

```diff
@@ -0,0 +1,629 @@
+                                       published_date = datetime.datetime.utcnow(),
+                                       version = version,
+                                       compatibility = release_old.compatibility)
+
+        upload_pbw(release_new, request.files['pbw'])
+        db.session.commit()
+
+        announce_release(app, release_new)
+
+        return jsonify(success = True)
+        
+    except Exception as e:
+        traceback.print_exc()
+        print("Oh no")
+        abort(500)
+
+# Screenshots 
+@devportal_api.route('/app/<appID>/screenshots')
+def missing_platform(appID):
+    return jsonify(error = "Missing platform", e = "platform.missing", message = "Use /app/<id>/screenshots/<platform>"), 400
+    
+@devportal_api.route('/app/<appID>/screenshots/<platform>', methods=['GET'])
+def get_app_screenshots(appID, platform):
+    # Check app exists
+
+    if not is_valid_platform(platform):
+        return jsonify(error = f"Invalid platform: {platform}", e = "platform.invalid"), 400  
+
+    app = App.query.filter(App.id == appID)
+    if app.count() < 1:
+        return jsonify(error = "Unknown app", e = "app.notfound"), 400      
+    app = app.one()
+
+    asset_collection = AssetCollection.query.filter(AssetCollection.app_id == app.id, AssetCollection.platform == platform).one_or_none()
+
+    if asset_collection is None:
+        return jsonify([])
+    else:
+        return jsonify(asset_collection.screenshots)
+
+@devportal_api.route('/app/<appID>/screenshots/<platform>', methods=['POST'])
+def new_app_screenshots(appID, platform):
+    app = App.query.filter(App.id == appID)
+    if app.count() < 1:
+        return jsonify(error = "Unknown app", e = "app.notfound"), 400
+    app = app.one()
+
+    # Check we own the app
+    result = authed_request('GET', f"{config['REBBLE_AUTH_URL']}/api/v1/me/pebble/appstore")
+    if result.status_code != 200:
+        abort(401)
+    me = result.json()
+    if not me['id'] == app.developer_id:
+        return jsonify(error = "You do not have permission to modify that app", e = "permission.denied"), 403
+
+    if not is_valid_platform(platform):
+        return jsonify(error = f"Invalid platform: {platform}", e = "platform.invalid"), 400  
+
+    asset_collection = AssetCollection.query.filter(AssetCollection.app_id == app.id, AssetCollection.platform == platform).one_or_none()
+
+    # Get the first image, this is a single image API
+    new_image = next(iter(request.files.to_dict()))
+    new_image = request.files[new_image]
+
+    # Check it's a valid image file
+    if not is_valid_image_file(new_image):
+        return jsonify(error = "Illegal image type", e = "screenshots.illegalvalue"), 400
+
+    if asset_collection is None:
+        asset_collection = clone_asset_collection_without_images(app, platform)
+        app.asset_collections[platform] = asset_collection
+    else:
+        # Check we don't already have 5 screenshots in this asset collection
```

Comment:
Elsewhere the limit is six. Maybe this should be a constant?

Reply:
well?

Reply:
It's consistently 5. There is a range(6) here https://github.com/pebble-dev/rebble-appstore-api/blob/devPortal2/appstore/developer_portal_api.py#L140 that should be range(1,6) because we want to go from 1 to 5 inclusive.